core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/GoogleAuthResource.scala

```diff
@@ -113,7 +118,8 @@ class GoogleAuthResource {
   def login(request: GoogleUserLoginRequest): TokenIssueResponse = {
     retrieveUserByGoogleAuthCode(request.authCode) match {
       case Success(user) =>
-        TokenIssueResponse(jwtToken(jwtClaims(user, dayToMin(TOKEN_EXPIRE_TIME_IN_DAYS))))
+        if (user.getPermission == 0) throw new NotAuthorizedException("Account pending approval.")
+        else TokenIssueResponse(jwtToken(jwtClaims(user, dayToMin(TOKEN_EXPIRE_TIME_IN_DAYS))))
```

Comment:
We should only do authentication here. permission/role is for authorization which is not part of login/token issuing logic.